src/main/kotlin/app/revanced/patches/youtube/layout/miniplayer/MiniplayerPatch.kt

```diff
@@ -0,0 +1,356 @@
+import com.android.tools.smali.dexlib2.iface.instruction.ReferenceInstruction
+import com.android.tools.smali.dexlib2.iface.instruction.TwoRegisterInstruction
+import com.android.tools.smali.dexlib2.iface.reference.FieldReference
+import com.android.tools.smali.dexlib2.iface.reference.TypeReference
+import com.android.tools.smali.dexlib2.immutable.ImmutableMethod
+import com.android.tools.smali.dexlib2.immutable.ImmutableMethodParameter
+
+// YT uses "Miniplayer" without a space between 'mini' and 'player.
+@Patch(
+    name = "Miniplayer",
+    description = "Adds options to change the in app minimized player, " +
+            "and if patching target 19.16+ adds options to use modern miniplayers.",
+    dependencies = [
+        IntegrationsPatch::class,
+        SettingsPatch::class,
+        AddResourcesPatch::class,
+        MiniplayerResourcePatch::class
+    ],
+    compatiblePackages = [
+        CompatiblePackage(
+            "com.google.android.youtube", [
+                "18.32.39",
+                "18.37.36",
+                "18.38.44",
+                "18.43.45",
+                "18.44.41",
+                "18.45.43",
+                "18.48.39",
+                "18.49.37",
+                "19.01.34",
+                "19.02.39",
+                "19.03.36",
+                "19.04.38",
+                "19.05.36",
+                "19.06.39",
+                "19.07.40",
+                "19.08.36",
+                "19.09.38",
+                "19.10.39",
+                "19.11.43",
+                "19.12.41",
+                "19.13.37",
+                // 19.14 is left out, as it has incomplete miniplayer code and missing some UI resources.
+                // It's simpler to not bother with supporting this single old version.
+                // 19.15 has a different code for handling sub title texts,
+                // and also probably not worth making changes just to support this single old version.
+                "19.16.39" // Earliest supported version with modern miniplayers.
+            ]
+        )
+    ]
+)
+@Suppress("unused")
+object MiniplayerPatch : BytecodePatch(
+    setOf(GetFormFactorFingerprint,
+        MiniplayerDimensionsCalculatorParentFingerprint,
+        MiniplayerResponseModelSizeCheckFingerprint,
+        MiniplayerOverrideFingerprint,
+        MiniplayerModernConstructorFingerprint,
+        MiniplayerModernViewParentFingerprint,
+        YouTubePlayerOverlaysLayoutFingerprint
+    )
+) {
+    private const val INTEGRATIONS_CLASS_DESCRIPTOR = "Lapp/revanced/integrations/youtube/patches/MiniplayerPatch;"
+
+    override fun execute(context: BytecodeContext) {
+        AddResourcesPatch(this::class)
+
+        // Modern mini player is only present and functional in 19.15+
+        // Of note, some modern miniplayer code is present in 19.14 but the feature is not complete.
+        val isPatchingLegacy = ytOutlinePictureInPictureWhite24 < 0
```

Comment:
Maybe a better name would be:
```suggestion
        // Of note, some modern miniplayer code is present in 19.14 but the feature is not complete.
        val isPatchingOldVersion = ytOutlinePictureInPictureWhite24 < 0
```